gcovr/writer/html.py

```diff
@@ -233,33 +236,46 @@ def set_coverage(self, covdata: CovData) -> None:
         self.decisions = dict_from_stat(stats.decision, self._coverage_to_class)
         self.calls = dict_from_stat(stats.call, self._coverage_to_class)
 
-    def add_file(self, cdata, link_report, cdata_fname):
-        stats = SummarizedStats.from_file(cdata)
+    def clear_files(self) -> None:
+        self.files = []
+
+    def add_file(
+        self, cdata: Union[DirectoryCoverage, FileCoverage], link_report, cdata_fname
+    ) -> None:
+        stats = (
+            cdata.stats
+            if isinstance(cdata, DirectoryCoverage)
+            else SummarizedStats.from_file(cdata)
+        )
 
         lines = {
             "total": stats.line.total,
             "exec": stats.line.covered,
+            "uncovered": stats.line.uncovered,
```

Comment:
This is needed now that Jinja is sorting.

Reply:
Why moving the sorting to Jinja level? Prefer to do this in python as for all writers.

Reply:
Because we cannot sort at the time we need to, for comparing statistics of files with directories, when directories store an aggregate of files underneath it.

Why prefer this in python? Why not do this in Jinja for all writers. In this PR, we pass in to the template the field we sort on and the reverse-or-nonreverse sort approach and we can do it all. Jinja knows all the data and can sort, so why do you prefer python, more than just "that's the way we always do it." ??

Note there's still an issue with the relative-path used with the txt writer. See "Note" I left in the linked PR, that we should just make a list of datums, not worry about sorting in Python, just worry about the tabular data, and have jinja do the sort.
https://github.com/gcovr/gcovr/pull/685

Reply:
Another reason: We are adding either DirectoryCoverage or we are adding FileCoverage. Those datas come from different datasets that we cannot sort them in one place, lest we create a more complex keying system. We now must call add_row with the different types in a place that isn't sorted:
https://github.com/macetw/gcovr/blob/45cb8e5c6fc94aa789ff909972d8fd73a3f3465e/gcovr/writer/html.py#L658

We could add a lambda sort here, going over this list. This feels more complicated to me.

Reply:
This would be needed for sorting at Jinja level.

Reply:
FYI, another likely feature request is to sort dynamically by the user in Javascript (or similar), where a user could sort on the UI by any field.

Reply:
In this case the sort must be implemented in the javascript and not in the jinja2 template. This has to be checked when this feature is going to be implemented.